### DISCUSSION TITLE
the server responded with a status of 404 (Not Found) for the frist modal popup showed

### DISCUSSION THREAD

**Discussion by AECSTacconi** [Blazored/Modal#352] [Q&A]
I have an app published on an IIS server. This app contains various Blazored modal popup that are working well but the first popup showed is not showed correctly. In the console there is the error:

> Failed to load resource: the server responded with a status of 404 (Not Found) blazored-modal.css:1 

Probably Blozored is not yet "loaded" when i call modal.show and that return the error.
Can some one help me to solve this problem?

Thanks

**Comment by chrissainty** [comment]
Can you provide some code showing what you've done. It's impossible to help without some context.

**Reply by AECSTacconi** [reply]
All is setted properly:

Configuredservices:

```
public void ConfigureServices(IServiceCollection services)
    {
        services.AddRazorPages();
        services.AddServerSideBlazor();
        services.AddSingleton<WeatherForecastService>();
        services.AddBlazoredModal();
    }
```
_Imports:

```
@using Blazored.Modal
@using Blazored.Modal.Services
```
App.razor:

```
<CascadingBlazoredModal>
<Router AppAssembly="@typeof(Program).Assembly">
    <Found Context="routeData">
        <RouteView RouteData="@routeData" DefaultLayout="@typeof(MainLayout)" />
    </Found>
    <NotFound>
        <LayoutView Layout="@typeof(MainLayout)">
            <p>Sorry, there's nothing at this address.</p>
        </LayoutView>
    </NotFound>
</Router>
```
_Host.cshtml

```
@page "/"
@namespace StampaManualeEtichetteAggiuntive.Pages
@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
@{
Layout = null;
}

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0" />
<link href="_content/Blazored.Modal/blazored-modal.css" rel="stylesheet" />
<title>StampaManualeEtichette</title>
<base href="~/" />
<link rel="stylesheet" href="css/bootstrap/bootstrap.min.css" />
<link href="css/site.css" rel="stylesheet" />
</head>
<body>
<app>
    <component type="typeof(App)" render-mode="ServerPrerendered" />
</app>

<div id="blazor-error-ui">
    <environment include="Staging,Production">
        An error has occurred. This application may no longer respond until reloaded.
    </environment>
    <environment include="Development">
        An unhandled exception has occurred. See browser dev tools for details.
    </environment>
    <a href="" class="reload">Reload</a>
    <a class="dismiss">🗙</a>
</div>

<script src="_framework/blazor.server.js"></script>
<script src="_content/Blazored.Modal/blazored.modal.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font- 
awesome/5.15.1/css/all.min.css">
<script src="css/font-awesome/js/all.min.js"></script>
</body>
</html>
```
And in my Index.razor

`[CascadingParameter] public IModalService Modal { get; set; }`

And in the OnInitialized i call visualize_popUP
```
public async Task visualize_popUP()
    {
        await Task.Delay(500);
        await Task.Yield();
        Modal.Show<Confirm>("Attenzione");
        await Task.Yield();
    }
```

**Reply by chrissainty** [reply]
I can't see any issues with your setup. This looks to be a local issue with your environment. I would suggest clearing your cache and trying the app again. It also might be worth looking in the dev tools and seeing there are any issues showing up there.

**Reply by AECSTacconi** [reply]
If the app is in IIS express ( with visual studio ) is working correctly. It only appen when i publish app in to an IIS server.
If i continue using app and show another modal popup ( loaded correctly ) if i refresh the page and visualize the one that was failing this time is showed without problem.
I think it's probably somenthing with loading time of blazored library.

**Reply by chrissainty** [reply]
It's not to do with the loading of the library. The CSS file is referenced in the host page and is present long before the library loads. Have you checked the dev tools? If you look at the network tab you should be able to see what is happening and what files are loaded when.

**Reply by AECSTacconi** [reply]
I re-publish the app to IIS server after some modification ( to other part of the software) and now all is working good.
If the problem will show again i will post the screen of the dev tools.

Thank you

**Reply by AECSTacconi** [reply]
Hi,
here is a screen of the network tab on the dev tools
![image](https://user-images.githubusercontent.com/76946621/139453203-d7dde647-14b1-4d10-ab3b-a19d255b3147.png)
